src/System.Text.Json/tests/Serialization/EnumTests.cs

```diff
@@ -33,6 +35,72 @@ public static class EnumTests
                 @"""MyUInt64Enum"" : " + ulong.MaxValue +
                 @"}";
 
+        private static readonly string s_jsonByteEnum =
+                @"{" +
+                @"""MyByteEnum"" : " + byte.MaxValue +
+                @"}";
+
+
+        public static IEnumerable<object[]> JsonWithInvalidOutOfBoundNumber
+        {
+            get
+            {
+                const string InvalidNumber = "18446744073709551616"; // ulong.MaxValue + 1;
+                foreach(var i in new dynamic[] { InvalidNumber, ulong.MaxValue })
+                {
+                    yield return new object[] { $"{{ \"MyEnum\" : \"{i}\" }}" };
+                    yield return new object[] { $"{{ \"MyEnum\" : {i} }}" };
+                    yield return new object[] { $"{{ \"MyByteEnum\" : \"{i}\" }}" };
+                    yield return new object[] { $"{{ \"MyByteEnum\" : {i} }}" };
+                    yield return new object[] { $"{{ \"MyUInt32Enum\" : \"{i}\" }}" };
+                    yield return new object[] { $"{{ \"MyUInt32Enum\" : {i} }}" };
+                }
+                foreach(var i in new dynamic[] { InvalidNumber })
```

Comment:
I don't think we need a `foreach` for a single element. Also, since it is a single type, we don't need `dynamic`.